### PR TITLE
Add root_hints for 740xd G-CL

### LINKED ISSUE #411: Overcloud deployment fails on R740xd-CL-G systems in Alias lab
During overcloud deployment on R740xd-CL-G systems in Alias lab, the following error occurs.
```
 [overcloud.Controller]: CREATE_COMPLETE  state changedWaiting for messages on queue 'tripleo' with no timeout.
Timed out waiting for port 22 from <controller_ip>
```
Setting root_hints for the nodes seems to fix the issue.
`openstack baremetal node set <UUID> --property root_device='{"model": "PERC H740P Adp"}'`
We need to include this in the playbooks.

### PR DESCRIPTION
**Pull request by asyedham**

### CODE REVIEW

**Comment by masco** [comment]
If i have mix of systems how this will behave?

**Comment by asyedham** [comment]
@masco I have added a check to ensure root_hints is only enabled on 740xd's on rack e26 and skips setting root_hints for the rest of 740xd's

**Comment by sanjaychari** [comment]
This PR gives the error
```
fatal: [undercloud]: FAILED! => {
    "msg": "The conditional check 'lab_name == 'alias' and machine_type == '740xd' and instackenv_content.nodes[0]['pm_addr'].split('.')[0].split('-')[1] == 'e26'' failed. The error was: error while evaluating conditional (lab_name == 'alias' and machine_type == '740xd' and instackenv_content.nodes[0]['pm_addr'].split('.')[0].split('-')[1] == 'e26'): 'machine_type' is undefined\n\nThe error appears to be in '/var/lib/jenkins/workspace/OSP-API-tester/jetpack/introspect.yml': line 47, column 9, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n      - name: enable root_hints(740xd G-CL)\n        ^ here\n"
}
```
I think this is because we are trying to use the variable `machine_type` which is set on localhost, but not on undercloud.

**Comment by sanjaychari** [comment]
Now getting the error
```
fatal: [undercloud]: FAILED! => {
    "msg": "The task includes an option with an undefined variable. The error was: 'node_uuids' is undefined\n\nThe error appears to be in '/var/lib/jenkins/workspace/OSP-API-tester/jetpack/tasks/set_root_hints.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- name: using introspection data to set root device\n  ^ here\n"
}
```

File: introspect.yml [hunk 1]
```diff
@@ -43,3 +43,7 @@
       - name: check and delete introspection failed nodes
         include_tasks: tasks/delete_introspection_failed_nodes.yml
         when: scale_compute_vms == false
+
+      - name: enable root_hints(740xd G-CL)
+        include_tasks: tasks/set_root_hints.yml
+        when: lab_name == 'alias' and hostvars['localhost']['machine_type'] == '740xd' and hostvars['localhost']['instackenv_content'].nodes[0]['pm_addr'].split('.')[0].split('-')[1] == 'e26'
```
Comment (masco) [inline introspect.yml]: this playbook will be invoked only if the first node of instack_env in e26 rack right? or am I missing something?

Comment (asyedham) [reply introspect.yml]: @masco Done

**Comment by sanjaychari** [comment]
@asyedham Chris had mentioned that racks e26-e29 contain G-CL servers. So maybe this PR should check for the range e26-e29 instead of only e26?

**Comment by asyedham** [comment]
> @asyedham Chris had mentioned that racks e26-e29 contain G-CL servers. So maybe this PR should check for the range e26-e29 instead of only e26?

@sanjaychari Let's not add the logic now without testing it on machines from e29 rack,

**Comment by sanjaychari** [comment]
> > @asyedham Chris had mentioned that racks e26-e29 contain G-CL servers. So maybe this PR should check for the range e26-e29 instead of only e26?
> 
> @sanjaychari Let's not add the logic now without testing it on machines from e29 rack,

@asyedham Ack

**Comment by sanjaychari** [comment]
Facing this error with the latest changes.
```
TASK [set fact for e26 rack machines in instackenv] ****************************
task path: /var/lib/jenkins/workspace/OSP-API-tester@2/jetpack/introspect.yml:47
Wednesday 17 November 2021  14:03:48 +0000 (0:00:00.056)       1:44:35.506 **** 
fatal: [undercloud]: FAILED! => {
    "msg": "'instackenv_content' is undefined"
}
```

**Comment by sanjaychari** [comment]
> Facing this error with the latest changes.
> 
> ```
> TASK [set fact for e26 rack machines in instackenv] ****************************
> task path: /var/lib/jenkins/workspace/OSP-API-tester@2/jetpack/introspect.yml:47
> Wednesday 17 November 2021  14:03:48 +0000 (0:00:00.056)       1:44:35.506 **** 
> fatal: [undercloud]: FAILED! => {
>     "msg": "'instackenv_content' is undefined"
> }
> ```

I think we might have to use `hostvars['localhost']['instackenv_content']` instead of `instackenv_content`?

**Comment by sanjaychari** [comment]
`ansible-playbook -vvv main.yml` deployed OSP 16.1 with this patch successfully on 740xd G-CL systems. I will trigger a CI build to try out deployment with tags as well.

File: introspect.yml [hunk 2]
```diff
@@ -39,7 +39,20 @@
...
       tn10rt: "{{ hostvars['localhost']['tn10rt'] }}"
+      instackenv_content: "{{ hostvars['localhost']['instackenv_content'] }}"
   tasks:
       - name: check and delete introspection failed nodes
         include_tasks: tasks/delete_introspection_failed_nodes.yml
         when: scale_compute_vms == false
+
+      - name: set fact for e26 rack machines in instackenv
+        set_fact:
+          rack: "{{ rack_info }}"
```
Comment (masco) [inline introspect.yml:50]: if the cloud doesn't have any machine from e26 rack, this will become undefined and cause an error at line number 58.

**Review by masco — CHANGES_REQUESTED** [review]

File: introspect.yml (hunk 2 above)
Comment (asyedham) [reply introspect.yml:50]: @masco fixed it

**Review by masco — APPROVED** [review]
LGTM

**Pull request merged**